.github/workflows/release.yml

```diff
@@ -0,0 +1,129 @@
+name: CLI Release
+
+on:
+  workflow_dispatch:
+  # schedule:
+  #   - cron: '0 12 * * 2'  # Runs every Tuesday at 12:00 PM (noon) ET
+
+jobs:
+
+  checks:
+    runs-on: ubuntu-latest
+    outputs:
+      HAS_TAG: ${{ steps.check_tag.outputs.has_tag }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      
+      - name: Check if commit has a new tag
+        id: check_tag
+        run: |
+          if git describe --tags --exact-match HEAD >/dev/null 2>&1; then
+            echo "has_tag=true" >> $GITHUB_OUTPUT
+            echo true
+          else
+            echo "has_tag=false" >> $GITHUB_OUTPUT
+            echo false
+          fi
+
+  release:
+    runs-on: ubuntu-latest
+    needs: checks
+    outputs:
+      CLI_VERSION: ${{ steps.get_cli_version.outputs.CLI_VERSION }}
+    if: needs.checks.outputs.HAS_TAG == 'false'
```

Comment:
This can be a trigger condition instead. Check out the docs:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore